src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,151 @@
+
+/**
+ * @author josefranciscodelvecchio
+ * Controller for CRUD User
+ */
+@RestController
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    /**
+     * This method find all users
+     * @return {@link List}<{@link User}>
+     */
+    @GetMapping
+    public List<User> findAll() {
+        return userRepository.findAll();
+    }
+
+    /**
+     * This method find users that match with title param
+     * @param userName: Title of the User
+     * @return {@link User}
+     */
+    @GetMapping("/user-name/{userName}")
+    public User findByUserName(@PathVariable String userName) {
+        return userRepository.findByUserName(userName)
+                .orElseThrow(UserNotFoundException::new);
+    }
+
+    /**
+     * This method find user by id param
+     * @param id: Id of the user
+     * @return {@link User}
+     * @throws UserNotFoundException when user not found
+     */
+    @GetMapping("/{id}")
+    public User findOne(@PathVariable Long id) {
+        return userRepository.findById(id)
+                .orElseThrow(UserNotFoundException::new);
+    }
+
+    /**
+     * This method creates an {@link User} with the data from user param
+     * @param user {@link User}: Data to create
+     * @return created {@link User}
+     */
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
+        return userRepository.save(user);
+    }
+
+    /**
+     * This method delete user by id param
+     * @param id: Id of the user
+     * @throws UserNotFoundException when user to delete not found
+     */
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable Long id) {
+        userRepository.findById(id)
+                .orElseThrow(UserNotFoundException::new);
+        userRepository.deleteById(id);
+    }
+
+    /**
+     * This method update data an user
+     * @param user
+     * @param id: Id of the user
+     * @return {@link User}: Data to update
+     * @throws UserIdMismatchException when id param and user's id mismatch
+     * @throws UserNotFoundException when user not found
+     */
+    @PutMapping("/{id}")
+    public User updateUser(@RequestBody User user, @PathVariable Long id) {
+        if (user.getId() != id) {
+            throw new UserIdMismatchException();
+        }
+        userRepository.findById(id)
+                .orElseThrow(UserNotFoundException::new);
+        return userRepository.save(user);
+    }
+
+    /**
+     * This method add an {@link Book} to user
+     * @param id: id of the user
+     * @param bookId: book id to add
+     * @return {@link User}
+     * @throws UserNotFoundException when user not found
+     * @throws BookNotFoundException when book to add not found
+     */
+    @PostMapping("/{id}/books/{bookId}")
+    @ResponseStatus(HttpStatus.CREATED)
```

Comment:
I'm not sure if this is the right status code, because you're not creating a resource..i'd use 200 (this is just a suggestion, you don't need to change it 😄 )

Reply:
i think any of two status code are valid, because http status doc say that 201 status is used when a resource is created independent if it's child or parent, but I'm not sure  😅